addons/info/src/components/markdown/pre/copyButton.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import glamorous, { withTheme } from 'glamorous';
+
+const Button = glamorous.button(
+  {
+    outline: 'none',
+    overflow: 'hidden',
+    border: '1px solid #eee',
+    borderRadius: 3,
+    backgroundColor: '#FFFFFF',
+    cursor: 'pointer',
+    fontSize: 13,
+    padding: '3px 10px',
+    alignSelf: 'flex-start',
+
+    ':hover': {
+      backgroundColor: '#f4f7fa',
+      borderColor: '#ddd',
+    },
+
+    ':active': {
+      backgroundColor: '#e9ecef',
+      borderColor: '#ccc',
+    },
+  },
+  props => props.styles
+);
+
+const ContentWrapper = glamorous.div(
+  {
+    transition: 'transform .2s ease',
+    height: 16,
+  },
+  props => ({
+    ...props.styles,
+    transform: props.toggled ? 'translateY(0px)' : 'translateY(-100%) translateY(-6px)',
+  })
+);
+
+function CopyButton(props) {
+  const {
+    copyButton: { toggleText = 'Copied!', text = 'Copy', ...copyButtonStyles } = {},
+    copyButtonContent,
+  } = props.theme;
```

Comment:
It's quite hard for me to wrap the head around nested destructuring assignments. Can you please make them no more than 1 level deep?